### ISSUE TITLE
Passing non-default `ImageProcessor` to `ImagePrefetcher` will always cause cache miss

### ISSUE THREAD

**Issue by Killectro** [onevcat/Kingfisher#589]
When you pass an `ImageProcessor` to `ImagePrefetcher` it will check to see if it exists in the cache like this:
```swift
let alreadyInCache = manager.cache.isImageCached(forKey: resource.cacheKey).cached
```

If that returns `false` it will continue on, download the image, process it, and store it like this: 

```swift
targetCache.store(image,
                  original: originalData,
                  forKey: key,
                  processorIdentifier:options.processor.identifier,
                  cacheSerializer: options.cacheSerializer,
                  toDisk: !options.cacheMemoryOnly,
                  completionHandler: nil)
```

Both cache keys are calculated using the same method:

```swift
let computedKey = key.computedKey(with: identifier)
```

but you might notice, that above when calling the `isImageCached(forKey:processorIdentifier:)` method, `ImagePrefetcher` doesn't pass the `processorIdentifier:` parameter. This means that if the URL is `example.com`, `ImagePrefetcher` is checking (the md5 hash of) `example.com` to see whether there is a cached version, but when it goes to download and store the image, it is storing it at `example.com@<processor_name>`. This means that the `ImageProcessor` will never have a cache hit if you pass anything other than `DefaultImageProcessor` as an option.

This can be remedied by simply passing the `processorIdentifier:` parameter as such:

```swift
let alreadyInCache = manager.cache.isImageCached(forKey: resource.cacheKey, processorIdentifier: optionsInfo.processor.identifier).cached
```

I will open a PR by EOD today that fixes this bug.

**Comment by onevcat** [comment]
You are definitely correct on this! Thank you!

**Comment by onevcat** [comment]
Should be fixed in version 3.4.0.
